debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java

```diff
@@ -145,10 +165,38 @@ public SnapshotResult<O> doExecute(ChangeEventSourceContext context, O previousO
             return SnapshotResult.completed(ctx.offset);
         }
         finally {
+            if (connectionPool != null) {
+                for (JdbcConnection conn : connectionPool) {
+                    if (!jdbcConnection.equals(conn)) {
```

Comment:
IMHO here should be identity comparison.

Reply:
@jpechane the equals method is the default one which does identity comparison. should we not trust the contract of the equals method?